src/main/java/seedu/duke/commands/AddCommand.java

```diff
@@ -0,0 +1,46 @@
+package seedu.duke.commands;
+
+import seedu.duke.exceptions.ModHappyException;
+import seedu.duke.parsers.AddParser;
+import seedu.duke.parsers.ModHappyParser;
+import seedu.duke.tasks.Task;
+
+import java.util.HashMap;
+
+
+public class AddCommand extends Command {
+
+    private static final String ADD_COMMAND_WORD = "add";
+    private static final String FLAG = "flag";
+    private static final String TASK_FLAG = "/t";
+    private static final String MOD_FLAG = "-mod";
+    private static final String ADD_MESSAGE_TOP = "Hey! I have added this task for you!\n";
+    private static String task = "task";
+    private static String mod = "mod";
+
+    public AddCommand(String arg) throws ModHappyException {
+        try {
+            commandName = ADD_COMMAND_WORD;
+            AddParser addParser = new AddParser();
+            HashMap<String, String> parsedArg = addParser.parseString(arg);
+            switch (parsedArg.get(FLAG)) {
+            case TASK_FLAG:
+                //add tasks
+                task = parsedArg.get("argument1");
+                Task.taskList.add(task);
+                break;
+            default:
+                throw new UnsupportedOperationException();
+            }
+        } catch (ModHappyException e) {
+            throw e;
+        }
+    }
+
```

Comment:
I suggest to put the execution of Command in execute method instead of constructor. In case we want to re-use the commands later